### PR TITLE
Fixed #28363 -- Allowed naming the migration generated by squashmigrations.

### PR DESCRIPTION
**Pull request by melvyn-sopacua**
https://code.djangoproject.com/ticket/28363

### CODE REVIEW

File: django/core/management/commands/squashmigrations.py [hunk 1]
```diff
@@ -32,6 +32,10 @@ def add_arguments(self, parser):
             '--noinput', '--no-input', action='store_false', dest='interactive', default=True,
             help='Tells Django to NOT prompt the user for input of any kind.',
         )
+        parser.add_argument(
+            '--finalname', dest='final_name', default=None, nargs=1,
+            help='Use this name for the resulting migration',
+        )
```
Comment (charettes) [inline django/core/management/commands/squashmigrations.py]: Any reasons why we can't reuse `--name, -name` from ` makemigations` instead?

Comment (melvyn-sopacua) [inline django/core/management/commands/squashmigrations.py]: No other then being explicit. There's 2 "names" in the command already.

Comment (timgraham) [inline django/core/management/commands/squashmigrations.py]: --file-name or --filename? Something like --squashed-name might be more intuitive.

**Comment by timgraham** [comment]
@melvyn-sopacua, do you plan to finish this as per, "A Trac ticket and documentation is also required. A Trac ticket and documentation is also required. Please use the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist)."?

**Comment by melvyn-sopacua** [comment]
Thanks for the reminder, Tim. I'll make the time this weekend.

**Comment by melvyn-sopacua** [comment]
Ticket: https://code.djangoproject.com/ticket/28363

**Comment by timgraham** [comment]
The docs go in `docs/ref/django-admin.txt` and a mention in `docs/releases/2.0.txt` is also appropriate. Please use the patch review checklist that I linked to above.

**Comment by melvyn-sopacua** [comment]
@timgraham Did you want the bit from the howto gone? I'm not convinced that's a good idea. Too much stuff is already tucked away only in refs.

**Pull request merged**